### PR TITLE
Add support for .heic images in fullscreen mode and show a proper error when the image fails to load

### LINKED ISSUE #911: show unsupported extension error rather than broken picture placeholder in image viewer
_No content_

>  Issue created by BlueBubbles Bot in the `BlueBubbles` commnity on behalf of `@jojejo` in `#support`.

### PR DESCRIPTION
**Pull request by tneotia**
Fixes #911.

Not the best solution because .heic will still appear slightly compressed in the fullscreen viewer but its better than not being able to see the image at all.

### CODE REVIEW

**Review by jjoelj — CHANGES_REQUESTED** [review]
* Change error text to "Failed to display image" rather than "Failed to load image"
* Use an if-else in `initBytes()` Don't need to use flutter's `readAsBytes()` for no reason
* make the text style of the error widget theme-reliant using a property of `Theme.of(context)`

**Comment by tneotia** [comment]
> * make the text style of the error widget theme-reliant using a property of `Theme.of(context)`

@jjoelj for that last one the viewer is black in light mode too so we don't want black text on black BG using `Theme.of(context)`. Is there some other property that I can use? 

Technically I could check if the color is black and correct it but it's not ideal.

**Review by jjoelj — APPROVED** [review]
> > * make the text style of the error widget theme-reliant using a property of `Theme.of(context)`
> 
> @jjoelj for that last one the viewer is black in light mode too so we don't want black text on black BG using `Theme.of(context)`. Is there some other property that I can use?
> 
> Technically I could check if the color is black and correct it but it's not ideal.

Now that I look at the existing code, it looks like a lot of the colors are hardcoded. In that case this looks good. We can open a separate ticket for making the photoviewer as a whole more responsive to the theme.

**Review by zlshames — APPROVED** [review]
Agreed, not a long term solution, but a step in the right direction

File: lib/layouts/image_viewer/image_viewer.dart [hunk 1]
```diff
@@ -273,6 +279,10 @@ class _ImageViewerState extends State<ImageViewer> with AutomaticKeepAliveClient
                       loadingBuilder: (BuildContext context, ImageChunkEvent ev) {
                         return loader;
                       },
+                      loadFailedChild: Center(
```
Comment (zlshames) [inline lib/layouts/image_viewer/image_viewer.dart:282]: Before I merge this, can you share what this widget will look like? Is it consistent with what we have in other areas?

**Comment by zlshames** [comment]
There is another ticket regarding .heic support. Since this is not a full solution, I am leaving that one open

**Pull request merged**